### PR TITLE
[build-script] Remove unused cmake options from Swift configuration.

### PR DESCRIPTION
**Pull request by rintaro**
#### What's in this pull request?

Before, `build-script-impl` emitted some unused or duplicated `-D` definitions.

Like:

```
-DCMAKE_INSTALL_PREFIX:PATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr \
-DLLVM_CONFIG:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/llvm-macosx-x86_64/bin/llvm-config \
-DCMARK_PATH_TO_CLANG_SOURCE:PATH=/Users/rintaro/git/swift-oss/llvm/tools/clang \
-DCMARK_PATH_TO_CLANG_BUILD:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/llvm-macosx-x86_64 \
-DCMARK_PATH_TO_LLVM_SOURCE:PATH=/Users/rintaro/git/swift-oss/llvm \
-DCMARK_PATH_TO_LLVM_BUILD:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/llvm-macosx-x86_64 \
-DCMARK_PATH_TO_CMARK_SOURCE:PATH=/Users/rintaro/git/swift-oss/cmark \
-DCMARK_PATH_TO_CMARK_BUILD:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/cmark-macosx-x86_64 \
-DCMARK_CMARK_LIBRARY_DIR:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/cmark-macosx-x86_64/src \
-DCMAKE_INSTALL_PREFIX:PATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr \
-DLLVM_CONFIG:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/llvm-macosx-x86_64/bin/llvm-config \
-DLLVM_PATH_TO_CLANG_SOURCE:PATH=/Users/rintaro/git/swift-oss/llvm/tools/clang \
-DLLVM_PATH_TO_CLANG_BUILD:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/llvm-macosx-x86_64 \
-DLLVM_PATH_TO_LLVM_SOURCE:PATH=/Users/rintaro/git/swift-oss/llvm \
-DLLVM_PATH_TO_LLVM_BUILD:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/llvm-macosx-x86_64 \
-DLLVM_PATH_TO_CMARK_SOURCE:PATH=/Users/rintaro/git/swift-oss/cmark \
-DLLVM_PATH_TO_CMARK_BUILD:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/cmark-macosx-x86_64 \
-DLLVM_CMARK_LIBRARY_DIR:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/cmark-macosx-x86_64/src \
-DCMAKE_INSTALL_PREFIX:PATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr \
-DLLVM_CONFIG:PATH=/Users/rintaro/git/swift-oss/build/Ninja-ReleaseAssert/llvm-macosx-x86_64/bin/llvm-config \
```

This PR removes unused `LLVM_` or `CMARK_` prefixed options, and duplicated `-DCMAKE_INSTALL_PREFIX` and `-DLLVM_CONFIG`.

Fixes this cmake warnings:

```
CMake Warning:
  Manually-specified variables were not used by the project:

    CMARK_CMARK_LIBRARY_DIR
    CMARK_PATH_TO_CLANG_BUILD
    CMARK_PATH_TO_CLANG_SOURCE
    CMARK_PATH_TO_CMARK_BUILD
    CMARK_PATH_TO_CMARK_SOURCE
    CMARK_PATH_TO_LLVM_BUILD
    CMARK_PATH_TO_LLVM_SOURCE
    LLVM_CMARK_LIBRARY_DIR
    LLVM_PATH_TO_CLANG_BUILD
    LLVM_PATH_TO_CLANG_SOURCE
    LLVM_PATH_TO_CMARK_BUILD
    LLVM_PATH_TO_CMARK_SOURCE
    LLVM_PATH_TO_LLVM_BUILD
    LLVM_PATH_TO_LLVM_SOURCE
```

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

### CODE REVIEW

**Comment by practicalswift** [comment]
@rintaro Very good! You are becoming a gardening hero :-)  Keep it up! :+1:

**Comment by practicalswift** [comment]
@modocache Do you have time to review? :-)

File: utils/build-script-impl [hunk 1]
```diff
@@ -1747,8 +1723,32 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_TOOLS_DEPLOYMENT_TARGETS[@]}"
...
+                    "${cmake_options[@]}"
+                    -DCMAKE_INSTALL_PREFIX:PATH="${INSTALL_PREFIX}"
+                    -DLLVM_CONFIG:PATH="$(build_directory_bin ${deployment_target} llvm)/llvm-config"
+                    -DSWIFT_PATH_TO_CLANG_SOURCE:PATH="${CLANG_SOURCE_DIR}"
+                    -DSWIFT_PATH_TO_CLANG_BUILD:PATH="${llvm_build_dir}"
+                    -DSWIFT_PATH_TO_LLVM_SOURCE:PATH="${LLVM_SOURCE_DIR}"
+                    -DSWIFT_PATH_TO_LLVM_BUILD:PATH="${llvm_build_dir}"
+                    -DSWIFT_PATH_TO_CMARK_SOURCE:PATH="${CMARK_SOURCE_DIR}"
+                    -DSWIFT_PATH_TO_CMARK_BUILD:PATH="$(build_directory ${deployment_target} cmark)"
+                )
```
Comment (modocache) [inline utils/build-script-impl]: I think this `cmake_options` assignment should be combined with the one above (the one that sets variables like `SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER`). Or is there a reason for the two assignments here?

**Comment by modocache** [comment]
Looks good to me, modulo one comment.

Could someone kick off the CI tests? Assuming those pass, I think this is good to merge.

File: utils/build-script-impl (hunk 1 above)
Comment (rintaro) [inline utils/build-script-impl]: Should be, but in this PR, the reason is to keep the output diff - of the order of the options -  mininum.

```
    ...
    -DSWIFT_ENABLE_LTO:BOOL=$(true_false "${SWIFT_ENABLE_LTO}")
    -DSWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER:BOOL=$(true_false "${BUILD_RUNTIME_WITH_HOST_COMPILER}")
    "${swift_cmake_options[@]}"
    -DCMAKE_INSTALL_PREFIX:PATH="${INSTALL_PREFIX}"
    -DLLVM_CONFIG:PATH="$(build_directory_bin ${deployment_target} llvm)/llvm-config"
    ...
```
is confusing for us.
More advanced re-arranging should be done in another commit, IMO.

**Comment by gottesmm** [comment]
@swift-ci Please test

**Comment by modocache** [comment]
The failing test that CI pointed out was fixed in https://github.com/apple/swift/commit/3aa4ff4332d4866fb27381d3c327743bcf801808. Unfortunately that failure prevented later tests from running, which I think would be relevant in this case (I'm personally interested in whether we can still install the build Swift products). Another "@swift-ci please test and merge" anybody?

**Comment by gottesmm** [comment]
@swift-ci Please test and merge

**Comment by rintaro** [comment]
Could someone re-run the test, since the another problem in the master was fixed in 88b70f15c10ca7b782442579df09e1990a874d40?

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Comment by gribozavr** [comment]
The failure is unrelated, merging.
